### PR TITLE
normalize the database

### PR DESCRIPTION
**Pull request by Esmatullah12**
In this project I worked on normalizing the database and improve the performance of some slow quires.
below
below is the screenshot of the slow quires details
Screenshot 2023-11-07 at 11 13 59
Screenshot 2023-11-07 at 11 18 18
Screenshot 2023-11-07 at 11 18 34

And here are the screen shot of the quires improving the performance
Screenshot 2023-11-07 at 16 15 40
Screenshot 2023-11-07 at 16 22 49
Screenshot 2023-11-07 at 16 28 06
fter improving the performance

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @Esmatullah12 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- This is a pair-programming activity and you ought to collaborate with your partner. If you have issues with collaboration, you can send a mail to _**Student Success**_ and they resolve it for you. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: schema.sql [hunk 1]
```diff
@@ -1,29 +1,53 @@
...
   PRIMARY KEY(vet_id, species_id), 
   FOREIGN KEY(vet_id) REFERENCES vets (id),
-  FOREIGN KEY (species_id) REFERENCES species (id));
+  FOREIGN KEY (species_id) REFERENCES species (id)
+);
+
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
 
+CREATE INDEX index_visits_animal_id ON visits(animal_id);
+CREATE INDEX idx_visits_vet_id ON visits(vet_id);
```
Comment (1GORDON) [inline schema.sql:52]: Great job on adding indexes 🤝
- However for this query `SELECT * FROM visits where vet_id = 2;` and according to the results in the output provided. There seems to be no reduction in the execution time but instead, the execution time increased. I would kindly recommend you remove the indexes and add them again to see if the execution time reduces.
- Before
before
- After
after

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Esmatullah12, and @Kingno3l ✋

Good job so far!

## Highlights!
- Added indexes ✔️ 
- Added screenshots of the results to the pull request description ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Esmatullah12) [reply schema.sql:52]: Remove the existing indexes from the visits table, and then reapply them to enhance performance. The screenshot below shows the results after removing and re-adding the indexes.
Screenshot 2023-11-08 at 09 53 45

**Review by V-Blaze — APPROVED** [review]
Hi @Esmatullah12, @Kingno3l  

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights
- Used indexes to improve query time✔✔
- Added screenshot before and after improvement :heavy_check_mark: 
- Correct git flow :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**